galsim/chromatic.py

```diff
@@ -367,6 +746,8 @@ def copy(self):
 
     # Helper function
     def _applyMatrix(self, J):
+        if hasattr(J, '__call__'):
+            self.noInterpolation()
```

Comment:
You are right, I moved the warning here.
Re: chromatic flux scaling, it depends on what we decide makes sense (see my earlier question).